ParabolInc/parabol
# feat(metrics): Enable Google Analytics client side page_view tracking

# Description

Fixes https://github.com/ParabolInc/parabol-data-sanctum/issues/450

## Demo

Screenshot of GA4 successfully tracked my `page_view` events and geo info (yes, I'm working by seaside now 😎)

<img width="1099" alt="Screenshot 2022-10-27 at 2 36 06 PM" src="https://user-images.githubusercontent.com/1879975/198403662-019bdf37-1989-4be6-8f23-03b3df7f2a23.png">

Here's another GA4 screenshot showing it successfully detected my userId:

<img width="1078" alt="Screenshot 2022-10-27 at 2 36 33 PM" src="https://user-images.githubusercontent.com/1879975/198404154-c40731fe-1263-4eb3-90e0-28518992b53f.png">


## Testing scenarios

- [ ] `page_view` events tracked
  - Add `GA_TRACKING_ID` to your `.env`. See [action-devops/environments/local](https://github.com/ParabolInc/action-devops/blob/master/environments/local#L72)
  - Start the server locally and go to any app page
  - Go to the Staging [GA4 realtime](https://analytics.google.com/analytics/web/#/p327845365/realtime/overview) dashboard and verify the `page_view` events come through

## Final checklist
